### PR TITLE
Include task tag in task telemetry

### PR DESCRIPTION
**Pull request by sfc-gh-rcheng**
Logging has a task tag that will also be added to each task telemetry event

### CODE REVIEW

File: src/main/java/com/snowflake/kafka/connector/internal/telemetry/SnowflakeTelemetryService.java [hunk 1]
```diff
@@ -18,6 +18,8 @@ public interface SnowflakeTelemetryService {
    */
   void setTaskID(String taskID);
 
+  void setTelemetryTag(String telemetryTag);
```
Comment (sfc-gh-japatel) [inline src/main/java/com/snowflake/kafka/connector/internal/telemetry/SnowflakeTelemetryService.java:21]: nit: java docs

File: src/main/java/com/snowflake/kafka/connector/SnowflakeSinkTask.java [hunk 2]
```diff
@@ -488,7 +490,7 @@ private KafkaRecordErrorReporter createKafkaRecordErrorReporter() {
     return result;
   }
 
-  private String getTaskLoggingTag() {
+  private String getTaskTag() {
```
Comment (sfc-gh-japatel) [inline src/main/java/com/snowflake/kafka/connector/SnowflakeSinkTask.java:493]: nit: shuold this name be more specific? `getKafkaConnectTaskTag`

File: src/main/java/com/snowflake/kafka/connector/internal/telemetry/SnowflakeTelemetryServiceFactory.java [hunk 3]
```diff
@@ -40,6 +40,15 @@ public SnowflakeTelemetryServiceBuilder setTaskID(String taskID) {
       return this;
     }
 
+    /**
```
Comment (sfc-gh-japatel) [inline src/main/java/com/snowflake/kafka/connector/internal/telemetry/SnowflakeTelemetryServiceFactory.java:43]: nit: one line comment here.

File: src/main/java/com/snowflake/kafka/connector/SnowflakeSinkTask.java [hunk 4]
```diff
@@ -148,6 +148,7 @@ private SnowflakeSinkService getSink() {
   @Override
   public void start(final Map<String, String> parsedConfig) {
     this.taskStartTime = System.currentTimeMillis();
+    String taskTag = this.getTaskTag();
```
Comment (sfc-gh-japatel) [inline src/main/java/com/snowflake/kafka/connector/SnowflakeSinkTask.java:151]: make this `final`

**Review by sfc-gh-japatel — APPROVED** [review]
couple of suggestions but otherwise lgtm, thanks! 

Also, google jfmt is failing, lets make sure it is fixed before pushing!

**Comment by sfc-gh-rcheng** [comment]
We don't currently have task level telemetry, only connector level telemetry. I believe task level telemetry should be added, but it is out of scope for this pr, so i'll abandon this and create a corresponding jira. thanks for the review though jay!

**Pull request closed without merging**